File: 프로그래머스/unrated/159994. 카드 뭉치/카드 뭉치.java
class Solution {
    public String solution(String[] cards1, String[] cards2, String[] goal) {
        String answer = "Yes";
        
        int OneIndex = 0;
        int TwoIndex = 0;
        for(int i=0; i<goal.length; i++) {
            if(OneIndex<cards1.length && goal[i].equals(cards1[OneIndex])) {
                OneIndex++;
            } else if(TwoIndex<cards2.length && goal[i].equals(cards2[TwoIndex])) {
                TwoIndex++;
            } else {
                answer = "No";
                break;
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.js
function solution(cards1, cards2, goal) {
    for(const str of goal) {
        if(cards1[0] === str) {
            cards1.shift();
        } else if(cards2[0] === str) {
            cards2.shift();
        } else {
            return "No";
        }
    }
    
    return "Yes";
}